### PR TITLE
Fix printing of tensors with imaginary expressions

### PR DESCRIPTION
**Pull request by aadya940**
* tensor
     *  Modified tensor coefficient variable such that it always spits out a numeric coefficient. So, no comparison between a non-numeric and zero arises. Hence, when printing a tensor expression involving a non numeric coefficient, sympy doesn't crashes.

### CODE REVIEW

**Comment by 1e9abhi1e10** [comment]
@aadya940 pls write the subheader from [here](https://github.com/sympy/sympy-bot/blob/master/sympy_bot/submodules.txt) or else write `NO ENTRY` 
Read the instructions for the .mailmap file [here](https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file)

File: .mailmap [hunk 1]
```diff
@@ -185,6 +185,8 @@
 #
 # See also: 'git shortlog --help' and 'git check-mailmap --help'.
 #
+
+Aadya Chinubhai <aadyachinubhai@gmail.com> Aadya Chinubhai <77720426+aadya940@users.noreply.github.com>
```
Comment (oscarbenjamin) [inline .mailmap:189]: After adding your name here you should run the `bin/mailmap_check.py` script which will reorder the lines in the .mailmap file.

Comment (oscarbenjamin) [reply .mailmap:189]: The file is required to be in alphabetical ordering to avoid merge conflicts.

**Comment by 1e9abhi1e10** [comment]
> As pointed out in Issue #24326.
> 
> _get_args_for_traditional_printer() was causing an error by compairing imaginary number to zero. Issue has been solved.
> 
> import sympy from sympy.tensor.tensor import TensorIndexType, TensorIndex, TensorHead
> 
> Lorentz = sympy.tensor.tensor.TensorIndexType('Lorentz', dummy_name='L') mu = sympy.tensor.tensor.TensorIndex('mu', Lorentz) A = sympy.tensor.tensor.TensorHead('A', [Lorentz])
> 
> sympy.pprint(sympy.I * A(mu))
> 
> Result:
> 
> mu

Follow the instructions for [Release Notes](https://github.com/sympy/sympy/wiki/Writing-Release-Notes)

File: sympy/tensor/tensor.py [hunk 2]
```diff
@@ -3629,7 +3629,7 @@ def __getitem__(self, item):
 
     def _get_args_for_traditional_printer(self):
         args = list(self.args)
-        if (self.coeff < 0) == True:
+        if (self.coeff.is_real and self.coeff < 0) == True:
```
Comment (oscarbenjamin) [inline sympy/tensor/tensor.py:3632]: As I said in the issue:
> A better fix would be to use `as_coeff_Mul` to split out a purely numeric coefficient.

**Comment by oscarbenjamin** [comment]
Neither the title of this PR nor the release note communicates to anyone else meaningfully what this is about. The title should be something like "Fix printing of tensors with imaginary expressions". The release note should explain what is fixed in a way that is relevant to users (who don't actually care what is in the code that you have written).

**Comment by sylee957** [comment]
I'm not sure if this can overlap with 
https://github.com/sympy/sympy/pull/24474

**Pull request closed without merging**